File: src/main/java/com/dodson/backendcoderockr/repository/UserRepository.java
package com.dodson.backendcoderockr.repository;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.dodson.backendcoderockr.domain.model.UserModel;

public interface UserRepository extends JpaRepository<UserModel, Long> {

        /**
         * Testing @Query.
         * Retrieve the user with the given UUID.
         * @param userId The userId to search for.
         * @return The {@link UserModel} or null if not found.
         */
        @Query("SELECT u FROM UserModel u WHERE userId = :userId")
        UserModel findUserByUserId(@Param("userId") UUID userId);

        /**
         * Retrieve the user with the given UUID.
         * @param userId The userId to search for.
         * @return The {@link UserModel} or null if not found.
         */
        UserModel findByUserId(UUID userId);
}

File: src/main/java/com/dodson/backendcoderockr/repository/UserRepository.kt
package com.dodson.backendcoderockr.repository

import com.dodson.backendcoderockr.domain.model.UserModel
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserRepository : JpaRepository<UserModel, Long> {
    /**
     * Retrieve the user with the given UUID.
     * @param userId The userId to search for.
     * @return The [UserModel] or null if not found.
     */
    fun findByUserId(userId: UUID?): UserModel?
}
